Systems/WeaponMasterySystem.cs

```diff
@@ -293,28 +287,24 @@ public static List<L10N.LocalisableString> GetSpellMasteryPrintableData(ulong st
 
             var spellPowerAdditionFactor = spellMastery.Mastery / 100 * spellMastery.Effectiveness;
 
-            var increasedSpellPowerStatString = L10N.Get(L10N.TemplateKey.MasteryAddedStat)
-                .AddField("{statType}", $"{Enum.GetName(UnitStatType.SpellPower)}")
-                .AddField("{value}", $"+{(float)AdditionalSpellPowerPer100PercentMastery * (float)spellPowerAdditionFactor:F2}");
-            var increasedSpellCriticalStrikeChanceStatString = L10N.Get(L10N.TemplateKey.MasteryAddedStat)
-                .AddField("{statType}", $"{Enum.GetName(UnitStatType.SpellCriticalStrikeChance)}")
-                .AddField("{value}", $"+{(float)AdditionalSpellCriticalStrikeChancePer100PercentMastery * (float)spellPowerAdditionFactor:F2}%");
-            var increasedSpellCriticalStrikeDamageStatString = L10N.Get(L10N.TemplateKey.MasteryAddedStat)
-                .AddField("{statType}", $"{Enum.GetName(UnitStatType.SpellCriticalStrikeDamage)}")
-                .AddField("{value}", $"+{(float)AdditionalSpellCriticalStrikeDamagePer100PercentMastery * (float)spellPowerAdditionFactor:F2}%");
-            var increasedSpellLifeLeechStatString = L10N.Get(L10N.TemplateKey.MasteryAddedStat)
-                .AddField("{statType}", $"{Enum.GetName(UnitStatType.SpellLifeLeech)}")
-                .AddField("{value}", $"+{(float)AdditionalSpellLifeLeechPer100PercentMastery * (float)spellPowerAdditionFactor:F2}%");
-            var increasedSpellCooldownRecoveryRateStatString = L10N.Get(L10N.TemplateKey.MasteryAddedStat)
-                .AddField("{statType}", $"{Enum.GetName(UnitStatType.SpellCooldownRecoveryRate)}")
-                .AddField("{value}", $"+{(float)AdditionalSpellCooldownRecoveryRatePer100PercentMastery * (float)spellPowerAdditionFactor:F2}%");
-            
-            result.Add(increasedSpellPowerStatString);
-            result.Add(increasedSpellCriticalStrikeChanceStatString);
-            result.Add(increasedSpellCriticalStrikeDamageStatString);
-            result.Add(increasedSpellLifeLeechStatString);
-            result.Add(increasedSpellCooldownRecoveryRateStatString);
-            
+            Database.MasteryStatConfig[MasteryType.Spell].ForEach(config =>
+            {
+                if (config.type == UnitStatType.SpellPower)
+                {
+                    var increasedStatString = L10N.Get(L10N.TemplateKey.MasteryAddedStat)
+                        .AddField("{statType}", $"{Enum.GetName(config.type)}")
+                        .AddField("{value}", $"+{config.rate * spellPowerAdditionFactor:F2}");
+                    result.Add(increasedStatString);
+                }
+                else
+                {
+                    var increasedStatString = L10N.Get(L10N.TemplateKey.MasteryAddedStat)
+                        .AddField("{statType}", $"{Enum.GetName(config.type)}")
+                        .AddField("{value}", $"+{config.rate * 100 * spellPowerAdditionFactor:F2}%");
```

Comment:
Why is this x100 when spell power is not? Why not just change the config for spell power to be 0.1 instead of 10?

Reply:
This is because spell power does not scale as a percentage. Instead, you need to specify the specific value of the increase. If I change the value in the config, I will need to perform this calculation when assigning the buff (and I prefer to omit the calculation there).